docs/src/main/asciidoc/security-openid-connect-web-authentication.adoc

```diff
@@ -191,6 +191,17 @@ In such cases a `quarkus.oidc.authentication.redirect-path` property has to be s
 
 If `quarkus.oidc.authentication.redirect-path` is set but the original request URL has to be restored after the user has been redirected back to a callback URL such as `http://localhost:8080/service/callback` then a `quarkus.oidc.authentication.restore-path-after-redirect` property has to be set to `true` which will restore the request URL such as `http://localhost:8080/service/1`, etc.
 
+[[oidc-cookies]]
+== Dealing with Cookies
+
+OIDC adapter uses cookies to keep the session, code flow and post logout state.
+
+If you access the protected resources with the overlapping or different roots, for example, `/index.html` and `/web-app/service` or `/web-app/service1` and `/web-app/service2` or `/web-app1/service` and `/web-app2/service` then most likely you also need to set a `quarkus.oidc.authentication.cookie-path` property to a path value common to all of the application resources, such as `/` or `/web-app`, etc.
+
+If your application is deployed across multiple domains then you may also need to set a `quarkus.oidc.authentication.cookie-domain` property.
+
+Otherwise the browser cache manager may keep request path specific cookies which in turn may lead to some difficult to diagnoze errors. For example, an authorization code flow may fail due to a missing state cookie if a user has initially accessed `/index.html` but configured a callback URI to `/web-app/callback`.
```

Comment:
```suggestion
If your application is deployed across multiple domains, make sure to set a `quarkus.oidc.authentication.cookie-domain` property, otherwise the browser cache manager may keep request path specific cookies. 
Not setting this property may lead to some errors that are difficult to diagnose, for example: an authorization code flow may fail due to a missing state cookie if a user has initially accessed `/index.html` but configured a callback URI to `/web-app/callback`.
```

Reply:
Hi @gastaldi Not sure about this last suggestion - the warning about the path specific cookies applies to the `path` property as well. I think I'd rather move `If your application is deployed across multiple domains then you may also need to set a `quarkus.oidc.authentication.cookie-domain` property.` to follow this warning which would be implied

Reply:
Hi @gastaldi I've updated the domain cookie text - as not setting it will affect the session cookie as opposed to a state cookie (which may be affected without a path property). How does it look ?

Reply:
I think it looks good. Good work :wink: